examples/webgl_loader_ldraw.html

```diff
@@ -244,6 +250,189 @@
+
+					newGeometry.setAttribute( 'position', new THREE.BufferAttribute( positions, 3 ) );
+					if ( normals !== null ) newGeometry.setAttribute( 'normal', new THREE.BufferAttribute( normals, 3 ) );
+
+					if ( isConditionalLine ) {
+
+						const controlArray0 = geometry.getAttribute( 'control0' ).array.subarray( vertStart, vertEnd );
+						const controlArray1 = geometry.getAttribute( 'control1' ).array.subarray( vertStart, vertEnd );
+						const directionArray = geometry.getAttribute( 'direction' ).array.subarray( vertStart, vertEnd );
+
+						newGeometry.setAttribute( 'control0', new THREE.BufferAttribute( controlArray0, 3, false ) );
+						newGeometry.setAttribute( 'control1', new THREE.BufferAttribute( controlArray1, 3, false ) );
+						newGeometry.setAttribute( 'direction', new THREE.BufferAttribute( directionArray, 3, false ) );
+
+					}
+
+					return newGeometry;
+
+				}
+
+				function addGeometry( mat, geometry, geometries ) {
+
+					const geoms = geometries[ mat.uuid ];
+					if ( ! geoms ) {
+						geometries[ mat.uuid ] = {
+							mat: mat,
+							arr: [ geometry ]
+						};
+					}
+					else geoms.arr.push( geometry );
+
+				}
+
+				function permuteAttribute( attribute, elemSize ) {
+
+					// Permutes first two vertices of each attribute element
+
+					if ( ! attribute ) return;
+
+					const verts = attribute.array;
+					const numVerts = Math.floor( verts.length / 3 );
+					let offset = 0;
+					for ( let i = 0; i < numVerts; i ++ ) {
+
+						const x = verts[ offset ];
+						const y = verts[ offset + 1 ];
+						const z = verts[ offset + 2 ];
+
+						verts[ offset ] = verts[ offset + 3 ];
+						verts[ offset + 1 ] = verts[ offset + 4 ];
+						verts[ offset + 2 ] = verts[ offset + 5 ];
+
+						verts[ offset + 3 ] = x;
+						verts[ offset + 4 ] = y;
+						verts[ offset + 5 ] = z;
+
+						offset += elemSize * 3;
+
+					}
+
+				}
+
+				// Traverse the object hierarchy collecting geometries and transforming them to world space
+
+				const meshGeometries = {};
+				const linesGeometries = {};
+				const condLinesGeometries = {};
+
+				object.updateMatrixWorld( true );
+				const normalMatrix = new THREE.Matrix3();
+
+				object.traverse( c => {
+
+					if ( c.isMesh | c.isLineSegments ) {
+
+						const elemSize = c.isMesh ? 3 : 2;
+
+						const matrixIsInverted = c.matrixWorld.determinant() < 0;
+						if ( matrixIsInverted ) {
+
+							permuteAttribute( c.geometry.attributes.position, elemSize );
+							permuteAttribute( c.geometry.attributes.normal, elemSize );
+
+						}
+
+						c.geometry.applyMatrix4( c.matrixWorld );
+
+						if ( c.isConditionalLine ) {
+
+							if ( matrixIsInverted ) {
+
+								permuteAttribute( c.geometry.attributes.control0, elemSize );
+								permuteAttribute( c.geometry.attributes.control1, elemSize );
+								permuteAttribute( c.geometry.attributes.direction, elemSize );
+
+							}
```

Comment:
I don't think theres any reason to permute the conditional lines, is there? The only reason the faces need to be adjusted is because of winding order which doesn't apply here.

Reply:
I was not sure about that, I'll change it.